### PR TITLE
add string functions

### PR DESCRIPTION
**Pull request by Aiee**
Add support for replace(), reverse(), split() from openCypher string functions

Updates:

functions below were added
substring()
toLower()
toString()
toUpper()

### CODE REVIEW

**Comment by jievince** [comment]
> Add support for replace(), reverse(), split() from openCypher string functions
> 
> TODOs:
> 
> * [ ]  substring()  currently have substr()
> * [ ]  toLower()     currently have lower()
> * [ ]  toString()
> * [ ]  toUpper()    currently have upper()

To define `subString`, you can add `functions_["subString"] = attr` at the end of the definition of `subStr`.

File: src/common/function/FunctionManager.cpp [hunk 1]
```diff
@@ -660,6 +664,59 @@ FunctionManager::FunctionManager() {
...
+            return Value::kNullBadType;
+        };
+    }
+    {
+        auto &attr = functions_["split"];
+        attr.minArity_ = 2;
+        attr.maxArity_ = 2;
+        attr.isPure_ = true;
+        attr.body_ = [](const auto &args) -> Value {
+            if (args[0].isStr(), args[1].isStr()) {
```
Comment (dutor) [inline src/common/function/FunctionManager.cpp:702]: Value of `args[0].isStr()` will be discarded.

File: src/common/function/FunctionManager.cpp [hunk 2]
```diff
@@ -660,6 +664,59 @@ FunctionManager::FunctionManager() {
...
+                std::string delim(args[1].getStr());
+                List res;
+                size_t pos = 0;
+                std::string token;
+
+                while ((pos = origStr.find(delim)) != std::string::npos) {
+                    token = origStr.substr(0, pos);
+                    res.emplace_back(token);
+                    origStr.erase(0, pos + delim.length());
+                }
```
Comment (dutor) [inline src/common/function/FunctionManager.cpp:713]: Consider to use `folly::split`

File: src/common/function/FunctionManager.cpp [hunk 3]
```diff
@@ -720,26 +827,53 @@ FunctionManager::FunctionManager() {
...
+                        if (start < 0) {
+                            LOG(ERROR) << "Invalid Start index " << start;
+                            return Value::kNullBadData;
+                        }
+                        if (start == 0) {
+                            return value;
+                        }
+                        return value.substr(start, length);
+                    }
+                    return Value::kNullBadType;
```
Comment (yixinglu) [inline src/common/function/FunctionManager.cpp:871]: merge these two cases to one by giving a default value to `length` according to args number.

File: src/common/function/FunctionManager.cpp [hunk 4]
```diff
@@ -660,6 +679,94 @@ FunctionManager::FunctionManager() {
...
+        attr.body_ = [](const auto &args) -> Value {
+             switch (args[0].type()) {
+                case Value::Type::NULLVALUE:
+                    return Value::kNullBadType;
+                case Value::Type::INT: {
+                    return std::to_string(args[0].getInt());
+                }
+                case Value::Type::FLOAT: {
+                    char f[24];
+                    snprintf(f, sizeof(f), "%.1f", args[0].getFloat());
```
Comment (yixinglu) [inline src/common/function/FunctionManager.cpp:745]: buggy when the float number are `10.001`

**Review by dutor — APPROVED** [review]
Well done.

**Review by yixinglu — APPROVED** [review]

**Pull request merged**